actionpack/lib/action_dispatch/journey/gtg/simulator.rb

```diff
@@ -22,18 +24,19 @@ def initialize(transition_table)
 
         def memos(string)
           input = StringScanner.new(string)
-          state = [0]
-          while sym = input.scan(%r([/.?]|[^/.?]+))
+          state = INITIAL_STATE
+
+          while sym = input.scan(MEMO_PATTERN)
             state = tt.move(state, sym)
           end
 
-          acceptance_states = state.find_all { |s|
-            tt.accepting? s
-          }
-
-          return yield if acceptance_states.empty?
+          acceptance_states = state.each_with_object([]) do |s, memos|
+            next unless tt.accepting? s
+            memo = tt.memo(s)
+            memos.concat(memo) unless memo.empty?
```

Comment:
Do we even need the empty check now? Presumably concat won't bother appending an empty array? Could inline `memo` if so.

Reply:
Always concatenating is a little slower than checking if the memo is empty, but I do prefer the way the code looks getting rid of the local variable. Should we just remove the check?

Reply:
As I investigated in the loop on running tests, all `memo` are non empty array, so I think the `memo.empty?` will not improve performance even for real world usage.